File: problems/001/p001.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();

    const result: u32 = resolve(3, 5, 1000);

    try stdout.print("Result={d}\n", .{result});
}

pub fn resolve(n1: u32, n2: u32, max: u32) u32 {
    var sum: u32 = 0;
    for (1..max) |i| {
        if (i % n1 == 0) {
            sum = sum + @as(u32, @intCast(i));
        } else if (i % n2 == 0) {
            sum = sum + @as(u32, @intCast(i));
        }
    }
    return sum;
}

File: problems/001/p001.rs
fn resolve(n1: u32, n2: u32, max: u32) -> u32 {
    let mut sum: u32 = 0;
    let mut i: u32 = n1;

    loop {
        if i >= max {
            break;
        }

        if i % n1 == 0 || i % n2 == 0 {
            sum += i;
        }

        i += 1;
    }

    return sum;
}

fn main() {
    let result = resolve(3, 5, 1000);
    println!("Result={}", result);
}
